bigframes/dataframe.py

```diff
@@ -937,6 +912,65 @@ def obj_ref_rt_to_html(obj_ref_rt) -> str:
         html_string += f"[{row_count} rows x {column_count} columns in total]"
         return html_string
 
+    def _get_anywidget_bundle(self, include=None, exclude=None):
+        """
+        Helper method to create and return the anywidget mimebundle.
+        This function encapsulates the logic for anywidget display.
+        """
+        from bigframes import display
+
+        # Process blob columns if needed
+        self._cached()
+        df = self.copy()
+        if bigframes.options.display.blob_display:
+            blob_cols = [
+                series_name
+                for series_name, series in df.items()
+                if series.dtype == bigframes.dtypes.OBJ_REF_DTYPE
+            ]
+            for col in blob_cols:
+                df[col] = df[col].blob._get_runtime(mode="R", with_metadata=True)
+
+        # Create and display the widget
+        widget = display.TableWidget(df)
+        widget_repr_result = widget._repr_mimebundle_(include=include, exclude=exclude)
+
+        # Handle both tuple (data, metadata) and dict returns
+        if isinstance(widget_repr_result, tuple):
+            widget_repr = dict(widget_repr_result[0])  # Extract data dict from tuple
+        else:
+            widget_repr = dict(widget_repr_result)
+
+        # Use deferred repr for text/plain of anywidget display.
+        # This ensures consistency with __repr__ and avoids unnecessary query execution
+        # when the user is just printing the last expression in a cell.
+        widget_repr["text/plain"] = formatter.repr_query_job(df._compute_dry_run())
+        widget_repr["text/html"] = self._repr_html_fallback_()
+        return widget_repr
+
+    def _repr_mimebundle_(self, include=None, exclude=None):
+        """
+        Custom display method for IPython/Jupyter environments.
+        This is called by IPython's display system when the object is displayed.
+        """
+        opts = bigframes.options.display
+        # Only handle widget display in anywidget mode
+        if opts.repr_mode == "anywidget":
+            try:
+                return self._get_anywidget_bundle(include=include, exclude=exclude)
+
+            except (AttributeError, ValueError, ImportError):
+                # Fallback: let IPython use _repr_html_fallback_() instead
+                warnings.warn(
+                    "Anywidget mode is not available. "
+                    "Please `pip install anywidget traitlets` or `pip install 'bigframes[anywidget]'` to use interactive tables. "
+                    f"Falling back to static HTML. Error: {traceback.format_exc()}"
+                )
+                # Don't return anything - let IPython fall back to _repr_html_fallback_()
+                pass
+
+        return {"text/html": self._repr_html_fallback_(), "text/plain": repr(self)}
```

Comment:
This would be calling more BQ API calls than necessary. Let's call to_pandas_batches() and fetch the first page, just like we do in the anywidget version.